boards/arm/mikroe_clicker_2/mikroe_clicker_2.dts

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright (c) 2020 Trifork
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/dts-v1/;
+#include <st/f4/stm32f407Xg.dtsi>
+
+/ {
+	model = "MikroE clicker 2 for STM32";
+	compatible = "mikroe,stm32-e407", "st,stm32f407";
+
+	chosen {
+		zephyr,console = &uart4;
+		zephyr,shell-uart = &uart4;
+		zephyr,sram = &sram0;
+		zephyr,flash = &flash0;
+		zephyr,ccm = &ccm0;
+	};
+
+
+	leds {
+		compatible = "gpio-leds";
+		ld1: led_1 {
+			gpios = <&gpioe 12 GPIO_ACTIVE_HIGH>;
+			label = "User LD1";
+		};
+		ld2: led_2 {
+			gpios = <&gpioe 15 GPIO_ACTIVE_HIGH>;
+			label = "User LD2";
+		};
+	};
+
+	gpio_keys {
+		compatible = "gpio-keys";
+		user_button_0: key_0 {
+			label = "Key PE0";
+			gpios = <&gpioe 0 GPIO_ACTIVE_LOW>;
+		};
+		user_button_1: key_1 {
+			label = "Key K1";
+			gpios = <&gpioe 10 GPIO_ACTIVE_LOW>;
+		};
+	};
+
+	aliases {
+		led0 = &ld1;
+		led1 = &ld2;
+		sw0 = &user_button_0;
+		sw1 = &user_button_1;
+	};
+};
+
+&usart2 {
+	current-speed = <9600>;
+	status = "okay";
+};
+
+&usart3 {
+	current-speed = <9600>;
+	status = "okay";
+};
+
+&uart4 {
+	current-speed = <115200>;
+	status = "okay";
+};
+
+&rtc {
+	status = "okay";
+};
+
+&usbotg_fs {
+	status = "okay";
+};
+
+
+&spi2 {
+	cs-gpios = <&gpioe 11 GPIO_ACTIVE_LOW>;
+	status = "okay";
+};
+
+&spi3 {
+	cs-gpios = <&gpioe 8 GPIO_ACTIVE_LOW>;
+	status = "okay";
+};
+
+&i2c2 {
+	status = "okay";
+};
+
+&i2c3 {
+	status = "okay";
+};
+
+&adc1 {
+	status ="okay";
```

Comment:
Would it be possible to have labels for the channels here ? I couldn't find (or understand) the documentation of it.
e.g.:

    mikrobus_1_adc: adc1_ch2 {
        channel = <2>;
    };
    mikrobus_2_adc: adc1_ch3 {
        channel = <3>;
    };

Reply:
That's not currently possible. We would need something along the line of the gpio-nexus for ADCs in order to support this...